lib/spot_build/buildkite_agent.rb

```diff
@@ -9,29 +9,46 @@ def initialize(token, org_slug)
       @org_slug = org_slug
     end
 
-    def the_end_is_nigh
-      return unless agent_running?
-      job = current_job
-      stop(true)
-      reschedule_job(job)
+    def the_end_is_nigh(host = Socket.gethostname)
+      agents = agents_on_this_host(host)
+      agents.each do |agent|
+        stop_agent(agent, force: true)
+      end
+      agents.each do |agent|
+        reschedule_job(agent.job)
+      end
+      agents.count
     end
 
-    def stop(force="false")
-      return unless agent_running?
-      @client.stop_agent(@org_slug, agent_id, "{\"force\": #{force}}")
+    def stop_agent(agent, force: false)
+      @client.stop_agent(@org_slug, agent.id, "{\"force\": #{force}}")
```

Comment:
```ruby
def stop_agent(org, id, options = {})
```

https://github.com/Shopify/buildkit/blob/master/lib/buildkit/client/agents.rb#L36

Reply:
Pretty sure I tried to use it without stringifying it and it didn't work.

Reply:
Yeah a string works for `data` in this call (although the variable is called `options` initially`) https://github.com/lostisland/sawyer/blob/master/lib/sawyer/agent.rb#L96